### PR TITLE
Update `resource.prlimit` for Python 3.12+

### PR DESCRIPTION
**Pull request by aminalaee**
https://docs.python.org/3.12/library/resource.html#resource.prlimit

### CODE REVIEW

File: stdlib/resource.pyi [hunk 1]
```diff
@@ -86,8 +85,5 @@ if sys.platform != "win32":
     def getrusage(__who: int) -> struct_rusage: ...
     def setrlimit(__resource: int, __limits: tuple[int, int]) -> None: ...
     if sys.platform == "linux":
-        @overload
-        def prlimit(pid: int, resource: int, limits: tuple[int, int]) -> tuple[int, int]: ...
-        @overload
-        def prlimit(pid: int, resource: int) -> tuple[int, int]: ...
+        def prlimit(__pid: int, __resource: int, __limits: tuple[int, int] | None = ...) -> tuple[int, int]: ...
```
Comment (AlexWaygood) [inline stdlib/resource.pyi:88]: According to the contributor in #4727, it's not possible to pass `None` to this function

```suggestion
        def prlimit(__pid: int, __resource: int, __limits: tuple[int, int] = ...) -> tuple[int, int]: ...
```

Comment (aminalaee) [reply stdlib/resource.pyi:88]: I tried that before, but fails:
```
error: resource.prlimit is inconsistent, runtime argument "limits" has a default value of type None, which is incompatible with stub argument type tuple[builtins.int, builtins.int]
```

Maybe it's changed in 3.12, I can check that.

Comment (AlexWaygood) [reply stdlib/resource.pyi:88]: Stubtest isn't always 100% correct on what's allowed at runtime. If necessary, we should keep the allowlist entry and add a comment saying that stubtest thinks `None` can be passed, but it's not actually allowed at runtime.

But yeah, I don't have a linux machine to test on, so please experiment and report back the results!! :D

Comment (aminalaee) [reply stdlib/resource.pyi:88]:
```shell
Python 3.11.4 (main, Jun  7 2023, 12:45:48) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import resource
>>> resource.prlimit(1, 2, None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not iterable

Python 3.12.0rc3+ (heads/3.12:538f505a37, Sep 28 2023, 14:02:27) [GCC 12.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import resource 
>>> resource.prlimit(1, 2, None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
PermissionError: [Errno 1] Operation not permitted
```

Comment (AlexWaygood) [reply stdlib/resource.pyi:88]: Okay, great, so it changed in 3.12. Probably needs to be this, then?

```suggestion
        if sys.version_info >= (3, 12):
            def prlimit(__pid: int, __resource: int, __limits: tuple[int, int] | None = ...) -> tuple[int, int]: ...
        else:
            def prlimit(__pid: int, __resource: int, __limits: tuple[int, int] = ...) -> tuple[int, int]: ...
```

**Review by AlexWaygood — APPROVED** [review]
Thanks! :D

**Pull request merged**